pinot-query-runtime/src/test/resources/queries/BooleanLogic.json

```diff
@@ -0,0 +1,106 @@
+{
+  "boolean_type": {
+    "tables": {
+      "bools": {
+        "schema": [{"name": "b", "type": "BOOLEAN"}],
+        "inputs": [[true], [false]]
+      }
+    },
+    "queries": [
+      {
+        "description": "should support booleans",
+        "sql": "select b FROM {bools}"
+      },
+      {
+        "description": "should support boolean literals",
+        "sql": "select b = true, b = false FROM {bools}"
+      }
+    ]
+  },
+  "boolean_implicit_casting": {
+    "tables": {
+      "tbl": {
+        "schema": [
+          {"name": "b", "type": "BOOLEAN"},
+          {"name": "i", "type": "INT"},
+          {"name": "d", "type": "DOUBLE"},
+          {"name": "s", "type": "STRING"},
+          {"name": "n", "type": "BIG_DECIMAL"},
+          {"name": "t", "type": "TIMESTAMP"},
+          {"name": "by", "type": "BYTES"}
+        ],
+        "inputs": [
+          [true, 1, 1.2, "foo", "1.234", "2022-01-02 03:45:00", "DEADBEEF"],
+          [true, 0, 0.0, "", "0.0", "2022-01-02 03:45:00", "00"],
+          [false, 1, 1.2, "foo", "1.234", "2022-01-02 03:45:00", "DEADBEEF"],
+          [false, 0, 0.0, "", "0.0", "2022-01-02 03:45:00", "00"]
+        ]
+      }
+    },
+    "queries": [
+      {
+        "note": "to be postgres compatible, this should throw an exception - but various other systems (like H2 and MySQL) support this",
+        "description": "check implicit cast between boolean and int",
+        "sql": "select b = i FROM {tbl}"
```

Comment:
can we add some function mix?
- boolean logic cascade: `(b = i) = true` 
- boolean logic combine `(b = i) AND/OR (b = n)`